scripts/rmc.py

```diff
@@ -93,19 +93,86 @@ def symbol_table_to_gotoc(json_filename, cbmc_filename, verbose=False, keep_temp
                                stdout=subprocess.PIPE,
                                stderr=subprocess.STDOUT)
-    if not quiet:
+    if output_to != None:
+        with open(output_to, "w") as f:
+            for line in iter(process.stdout.readline, ""):
+                f.write(line)
+    elif not quiet:
         for line in iter(process.stdout.readline, ""):
             print(line, end="")
     process.stdout.close()
     retcode = process.wait()
     print_rmc_step_status("cbmc", process, verbose)
     return retcode
 
+def run_visualize(cbmc_filename, cbmc_args, verbose=False, quiet=False, keep_temps=False):
+    results_filename = "results.xml"
+    coverage_filename = "coverage.xml"
+    property_filename = "property.xml"
+    temp_goto_filename = "temp.goto"
+    if not keep_temps:
+        for filename in [results_filename, coverage_filename, property_filename, temp_goto_filename]:
+            atexit.register(delete_file, filename)
+
+    """
+    1) run rmc to build to <cbmc_filename>
+    2) goto-cc --function main <cbmc_filename> -o temp.goto
+    3) cbmc --xml-ui --trace ~/rmc/library/rmc/rmc_lib.c temp.goto > results.xml
+    4) cbmc --xml-ui --cover location ~/rmc/library/rmc/rmc_lib.c temp.goto > coverage.xml
+    5) cbmc --xml-ui --show-properties ~/rmc/library/rmc/rmc_lib.c temp.goto > property.xml
+    6) cbmc-viewer --result results.xml --coverage coverage.xml --property property.xml --srcdir . --goto temp.goto
+    """
+
+    retcode = run_cbmc(cbmc_filename, cbmc_args, verbose, quiet)
+
+    run_goto_cc(cbmc_filename, temp_goto_filename, verbose, quiet)
+    
+    cbmc_xml_args = cbmc_args + ["--xml-ui"]
+    run_cbmc(temp_goto_filename, cbmc_xml_args + ["--trace"], verbose, quiet, results_filename)
+    run_cbmc(temp_goto_filename, cbmc_xml_args + ["--cover", "location"], verbose, quiet, coverage_filename)
+    run_cbmc(temp_goto_filename, cbmc_xml_args + ["--show-properties"], verbose, quiet, property_filename)
+
+    run_cbmc_viewer(temp_goto_filename, results_filename, coverage_filename, property_filename, verbose, quiet)
+
+    return retcode
+
+def run_goto_cc(src, dst, verbose=False, quiet=False):
+    cmd = ["goto-cc"] + ["--function", "main"] + [src] + ["-o", dst]
+    process = subprocess.Popen(cmd, universal_newlines=True,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
+    if not quiet:
+        for line in iter(process.stdout.readline, ""):
+            print(line, end="")
+    process.stdout.close()
+    retcode = process.wait()
+    print_rmc_step_status("goto-cc", process, verbose)
+    return retcode
+
+def run_cbmc_viewer(goto_filename, results_filename, coverage_filename, property_filename, verbose=False, quiet=False):
+    cmd = ["cbmc-viewer"] + \
+          ["--result", results_filename] + \
+          ["--coverage", coverage_filename] + \
+          ["--property", property_filename] + \
+          ["--srcdir", "."] + \
```

Comment:
Almost certainly the srcdir and wkdir need to be parameters.  For example, for our regression tests, srcdir is going to be the root of the rmc source tree (including the stdlib) and wkdir is going to be '.' the regression test directory.   And they both need to be expanded to physical (not logical) paths.

Reply:
I'll need some help understanding where to get these values from, then.

Reply:
Fixed